src/views/workflow-history-v2/hooks/use-workflow-history-fetcher.ts

```diff
@@ -0,0 +1,100 @@
+import { useCallback, useEffect, useRef } from 'react';
+
+import {
+  type InfiniteData,
+  type InfiniteQueryObserverResult,
+  useQueryClient,
+} from '@tanstack/react-query';
+
+import useThrottledState from '@/hooks/use-throttled-state';
+import { type GetWorkflowHistoryResponse } from '@/route-handlers/get-workflow-history/get-workflow-history.types';
+import { type RequestError } from '@/utils/request/request-error';
+import WorkflowHistoryFetcher from '@/views/workflow-history/helpers/workflow-history-fetcher';
+import { type ShouldContinueCallback } from '@/views/workflow-history/helpers/workflow-history-fetcher.types';
+
+import {
+  type UseWorkflowHistoryFetcherParams,
+  type UseWorkflowHistoryFetcherOptions,
+} from './use-workflow-history-fetcher.types';
+
+export default function useWorkflowHistoryFetcher(
```

Comment:
Any reason not to directly import the existing `use-workflow-history-fetcher.ts`?